lib/Injector.php

```diff
@@ -363,22 +363,28 @@ public function make($name, array $args = array())
             return $this->shares[$normalizedClass];
         }
 
-        if (isset($this->delegates[$normalizedClass])) {
-            $executable = $this->buildExecutable($this->delegates[$normalizedClass]);
-            $reflectionFunction = $executable->getCallableReflection();
-            $args = $this->provisionFuncArgs($reflectionFunction, $args);
-            $obj = call_user_func_array(array($executable, '__invoke'), $args);
-        } else {
-            $obj = $this->provisionInstance($className, $normalizedClass, $args);
-        }
+        try {
+            if (isset($this->delegates[$normalizedClass])) {
+                $executable = $this->buildExecutable($this->delegates[$normalizedClass]);
+                $reflectionFunction = $executable->getCallableReflection();
+                $args = $this->provisionFuncArgs($reflectionFunction, $args);
+                $obj = call_user_func_array(array($executable, '__invoke'), $args);
+            } else {
+                $obj = $this->provisionInstance($className, $normalizedClass, $args);
+            }
 
-        $obj = $this->prepareInstance($obj, $normalizedClass);
+            $obj = $this->prepareInstance($obj, $normalizedClass);
 
-        if (array_key_exists($normalizedClass, $this->shares)) {
-            $this->shares[$normalizedClass] = $obj;
-        }
+            if (array_key_exists($normalizedClass, $this->shares)) {
+                $this->shares[$normalizedClass] = $obj;
+            }
 
-        unset($this->inProgressMakes[$normalizedClass]);
+            unset($this->inProgressMakes[$normalizedClass]);
+        }
+        catch (\Exception $exception) {
```

Comment:
This should be catching `\Throwable`

Reply:
This library targets PHP >=5.3.0, at least for now.

Reply:
It should be catching both; as a fallback. An exception doesn't have to exist to be specified in a catch() block (the catch block will just only be ever entered if the exception does exist).

Reply:
@Furgas Right, been a while since I've written 5.x compatible code. You can still support 7 with the following catch blocks:

```php
} catch (\Throwable $exception) {
    // Executed in PHP 7+
} catch (\Exception $exception) {
    // Executed in PHP 5.3 to 5.6
}
```

Reply:
Nice, TIL. Pushed the change.